game/src/ecs/systems/CollisionSystem.java

```diff
@@ -9,63 +9,52 @@
-                                                                                        hitbox2);
-                                                                try {
-                                                                    ((HitboxComponent) hitbox1)
-                                                                            .collide(
-                                                                                    (HitboxComponent)
-                                                                                            hitbox2,
-                                                                                    d);
-                                                                } catch (
-                                                                        InvocationTargetException
-                                                                                e) {
-                                                                    throw new RuntimeException(e);
-                                                                } catch (IllegalAccessException e) {
-                                                                    throw new RuntimeException(e);
-                                                                }
-                                                                try {
-                                                                    ((HitboxComponent) hitbox2)
-                                                                            .collide(
-                                                                                    (HitboxComponent)
-                                                                                            hitbox1,
-                                                                                    inverse(d));
-                                                                } catch (
-                                                                        InvocationTargetException
-                                                                                e) {
-                                                                    throw new RuntimeException(e);
-                                                                } catch (IllegalAccessException e) {
-                                                                    throw new RuntimeException(e);
-                                                                }
-                                                            }
-                                                        });
-                                    }
-                                }
-                            });
+    private void checkForCollision(CSData csd1, CSData csd2) {
+        if (csd1.e != csd2.e && checkForCollision(csd1.hc, csd2.hc)) {
+            Tile.Direction d = checkDirectionOfCollision(csd1.hc, csd2.hc);
+            try {
```

Comment:
Die ganzen Exceptions sind glaub outdated, oder? Sieht mir nach Reflection aus und mittlerweile ist das doch alles Strategy-Pattern.

Reply:
ja die sind noch drinnen geblieben und habe sie in den umbau von immer collision zu onEnter und onLeave entfernt. gehört aber nicht hier in den PR

Reply:
> gehört aber nicht hier in den PR

Da haben recht.